osu.Game.Rulesets.Karaoke/Mods/ModStage.cs

```diff
@@ -37,6 +33,15 @@ public void ApplyToBeatmap(IBeatmap beatmap)
         // trying to create a new one if has no matched stage info.
         // it's ok to like it as null if is not able to create the default one, beatmap processor will handle that.
         karaokeBeatmap.CurrentStageInfo = matchedStageInfo ?? CreateStageInfo(karaokeBeatmap)!;
+
+        // should invalidate the working property here because the stage info is changed.
+        // has the same logic in the beatmap processor.
+        beatmap.HitObjects.OfType<Lyric>().ForEach(x =>
+        {
+            x.InvalidateWorkingProperty(LyricWorkingProperty.Timing);
+            x.InvalidateWorkingProperty(LyricWorkingProperty.EffectApplier);
+        });
+        beatmap.HitObjects.OfType<Note>().ForEach(x => x.InvalidateWorkingProperty(NoteWorkingProperty.EffectApplier));
```

Comment:
Remove this code if make the stage into individual namespace and give it something like `stage processor`.
We will move all the stage invalidate logic into there.